### PR TITLE
WIP: Vtol transition failsafe RTL

### PR DESCRIPTION
**Pull request by sanderux**
Fixes: https://github.com/PX4/Firmware/issues/3663
This publishes the failsafe state after a failed vtol front transition.
Commander handles the failsafe and sets RTL if in mission mode.

### CODE REVIEW

**Comment by AndreasAntener** [comment]
This looks good! I think you could add the timeout to the other types as well.

**Comment by sanderux** [comment]
@AndreasAntener I have updated the tailsitter to include the timeout. i had to change the transition abort handling for this. I tested in SITL and it looks good. 

The same changes should be required for tiltrotor but i have no way of testing this. Can i assume it will work identical to tailsitter?

**Comment by AndreasAntener** [comment]
It should work more or less the same for all types. So I think we'll be good. If it doesn't look clear enough we could also organize a quick bench test with the Firefly6.

File: src/modules/vtol_att_control/tailsitter.cpp [hunk 1]
```diff
@@ -137,13 +142,9 @@ void Tailsitter::update_vtol_state()
 			break;
 
 		case FW_MODE:
-			_vtol_schedule.flight_mode 	= TRANSITION_BACK;
```
Comment (AndreasAntener) [inline src/modules/vtol_att_control/tailsitter.cpp]: This looks wrong, why is this change necessary?

**Comment by sanderux** [comment]
@AndreasAntener it's not, it just looks weird in the diff
It now reads:

            case FW_MODE:
		case TRANSITION_FRONT_P1:
			_vtol_schedule.flight_mode 	= TRANSITION_BACK;
			_vtol_schedule.transition_start = hrt_absolute_time();			
			break;

Before it read:

               case FW_MODE:
			_vtol_schedule.flight_mode 	= TRANSITION_BACK;
			_vtol_schedule.transition_start = hrt_absolute_time();
			break;

		case TRANSITION_FRONT_P1:
			// failsafe into multicopter mode
			_vtol_schedule.flight_mode = MC_MODE;
			break;

When in TRANSITION_FRONT_P1 mode and canceling the flight it also needs to backt transition.

**Comment by AndreasAntener** [comment]
There are two different cases:
- regular back transition from FW_MODE
- instantaneous back transition from TRANSITION_FRONT_P1

Your changes removes the later.

**Comment by sanderux** [comment]
Yes, because this is required.
if https://github.com/PX4/Firmware/blob/master/src/modules/vtol_att_control/tailsitter.cpp#L335-L359 is not executed the vehicle will not level back to MC mode and crash.

I have SITL tested this, you can try by reverting that bit and sitl test by setting transition airspeed to 20m/s. the vehicle will crash.

**Comment by AndreasAntener** [comment]
This is why I asked. If that's the case it points to a problem in the failsafe backtransition we need to check before this (and not just changing the logic so it works). Let's just take the timeout handling for the standard vtol in then.

**Comment by AndreasAntener** [comment]
I'll do a test and pull the standard vtol timeout in today, then we have a look about the other types.

**Comment by sanderux** [comment]
When transition has started the _v_att_sp->pitch_body is changed.
This always needs to be changed back when failsafing to MC mode.
I believe that canceling a transition during P1 manually will now also fail

**Comment by sanderux** [comment]
rebased

**Comment by AndreasAntener** [comment]
I'll push a new version of this without the tailsitter update as soon as I can compile again...

**Comment by AndreasAntener** [comment]
@sanderux Closing this. Will push the other branch to master if checks pass. After that, could you open a new PR with your last commit? So we can also look for the problem with the failsafe back transition on the tailsitter.
